File: java/src/main/java/toolgood/flowVision/Flows/FactoryMachineWork.java
package toolgood.flowVision.Flows;

import com.alibaba.fastjson2.JSONObject;
import toolgood.algorithm2.math.mathParser;
import toolgood.flowVision.Engines.FlowEngine;

public class FactoryMachineWork {
    public String Factory;
    public String MachineCode;
    public String MachineName;
    public String MachineCategoryCode;
    public String MachineCategoryName;
    public String CheckFormula;
    private ProjectWork Project;

    final static FactoryMachineWork parse(JSONObject jsonObject) {
        FactoryMachineWork result = new FactoryMachineWork();
        result.Factory = jsonObject.getString("factory");
        result.MachineCode = jsonObject.getString("machineCode");
        result.MachineName = jsonObject.getString("machineName");
        result.MachineCategoryCode = jsonObject.getString("machineCategoryCode");
        result.MachineCategoryName = jsonObject.getString("machineCategoryName");
        result.CheckFormula = jsonObject.getString("checkFormula");

        return result;
    }

    public void Init(ProjectWork work) {
        Project = work;
    }

    public boolean Check(FlowEngine engine) throws Exception {
        if (CheckFormula == null || CheckFormula.equals("")) {
            return true;
        }
        mathParser.ProgContext progContext = Project.CreateProgContext(CheckFormula);
        return engine.TryEvaluate(progContext, false);
    }
}

File: csharp/ToolGood.FlowVision/Flows/FactoryMachineWork.cs
using System.Text.Json.Serialization;
using ToolGood.FlowVision.Engines;

namespace ToolGood.FlowVision.Flows
{
	public sealed class FactoryMachineWork
	{
		private ProjectWork Project;// { get; set; }

		public string Factory { get; set; }

		/// <summary>
		/// 编号
		/// </summary>
		public string MachineCode { get; set; }

		/// <summary>
		/// 名称
		/// </summary>
		public string MachineName { get; set; }

		/// <summary>
		/// 编号
		/// </summary>
		[System.Text.Json.Serialization.JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingDefault)]
		public string MachineCategoryCode { get; set; }

		/// <summary>
		/// 名称
		/// </summary>
		[System.Text.Json.Serialization.JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingDefault)]
		public string MachineCategoryName { get; set; }

		/// <summary>
		/// 检测公式
		/// </summary>
		[System.Text.Json.Serialization.JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingDefault)]
		public string CheckFormula { get; set; }

		internal void Init(ProjectWork work)
		{
			Project = work;
		}

		internal bool Check(FlowEngine engine)
		{
			if (string.IsNullOrEmpty(CheckFormula)) { return true; }
			var progContext = Project.CreateProgContext(CheckFormula);
			return engine.TryEvaluate(progContext, false);
		}
	}
}